yt_dlp/extractor/microsoftembed.py

```diff
@@ -63,3 +71,253 @@ def _real_extract(self, url):
+    _TESTS = [{
+        'url': 'https://learn.microsoft.com/en-us/shows/bash-for-beginners/what-is-the-difference-between-a-terminal-and-a-shell-2-of-20-bash-for-beginners/',
+        'info_dict': {
+            'id': 'd44e1a03-a0e5-45c2-9496-5c9fa08dc94c',
+            'ext': 'ismv',
+            'title': 'What is the Difference Between a Terminal and a Shell? (Part 2 of 20)',
+            'description': 'md5:7bbbfb593d21c2cf2babc3715ade6b88',
+            'timestamp': 1676339547,
+            'upload_date': '20230214',
+            'thumbnail': r're:https://learn\.microsoft\.com/video/media/.*\.png',
+            'subtitles': 'count:14',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        entry_id = self._html_search_meta('entryId', webpage, 'entryId', fatal=True)
+        video_info = self._download_json(
+            f'https://learn.microsoft.com/api/video/public/v1/entries/{entry_id}', video_id)
+        return {
+            'id': entry_id,
+            'formats': self._extract_ism(video_info['publicVideo']['adaptiveVideoUrl'], video_id),
+            'subtitles': self._sub_to_dict(traverse_obj(video_info, ('publicVideo', 'captions', ..., {
+                'tag': ('language', {str}),
+                'url': ('url', {url_or_none}),
+            }))),
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+            **traverse_obj(video_info, {
+                'timestamp': ('createTime', {parse_iso8601}),
+                'thumbnails': ('publicVideo', 'thumbnailOtherSizes', ..., {url_or_none}, {lambda x: x and {'url': x}}),
+            }),
+        }
+
+
+class MicrosoftLearnSessionIE(InfoExtractor):
+    _VALID_URL = r'https?://learn\.microsoft\.com/(?:[\w\-]+/)?events/[\w\-]+/(?P<id>[^?#/]+)'
+    _TESTS = [{
+        'url': 'https://learn.microsoft.com/en-us/events/build-2022/ts01-rapidly-code-test-ship-from-secure-cloud-developer-environments',
+        'info_dict': {
+            'id': '9640d86c-f513-4889-959e-5dace86e7d2b',
+            'ext': 'ismv',
+            'title': 'Rapidly code, test and ship from secure cloud developer environments - Events',
+            'description': 'md5:f26c1a85d41c1cffd27a0279254a25c3',
+            'timestamp': 1653408600,
+            'upload_date': '20220524',
+            'thumbnail': r're:https://mediusimg\.event\.microsoft\.com/video-\d+/thumbnail\.jpg.*',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        metainfo = {
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+            'timestamp': parse_iso8601(self._html_search_meta('startDate', webpage, 'startDate')),
```

Comment:
Original impl had `default=None`. Do we want to warn when a timestamp is not found? Or `default=None` again?
```suggestion
            'timestamp': parse_iso8601(
                self._html_search_meta('startDate', webpage, 'startDate', default=None)),
```

Reply:
A warn sounds like a better way to me. https://github.com/yt-dlp/yt-dlp/pull/9177/commits/8730880d2b0c17052f51bf3957d7d4c3dfdcbc22